neptune/new/metadata_containers/metadata_container.py

```diff
@@ -274,6 +276,12 @@ def define(
         elif is_dict_like(value):
             value = Namespace(value)
         elif is_string_like(value):
+            simple_warning(
+                "You're assigning an object with an implicit cast to a string."
+                " Use `... = str(object)` instead."
+                " We'll end support of this behavior in `neptune-client==1.0.0`.",
```

Comment:
@normandy7 Anything you would like to change here?

Reply:
Ah yeah, let me give this another pass. Wasn't sure exactly what to look at, so overlooked this the first time I went over the changed files.

Reply:
To make sure I understand correctly, what kinds of object would this warning show up for? The line `" Use `... = str(object)` instead."` suggests that logging the object as a string is the intention, but could it be that it's not?

Reply:
For now **any type** that is not explicitly supported is implicitly converted to `str`. This behaviour will be deprecated.

Reply:
Right. Would it be more clear if we do:
```
"The object you're logging will be implicitly cast to a string."
" We'll end support of this behavior in `neptune-client==1.0.0`.",
" To log the object as a string, use `... = str(object)` instead."
```

Reply:
Done